libs/wingc/src/type_check.rs

```diff
@@ -3879,7 +3888,14 @@ impl<'a> TypeChecker<'a> {
 		let (var, var_phase) = self.resolve_reference(&variable, env);
 
 		if !var.type_.is_unresolved() && !var.reassignable {
-			self.spanned_error(variable, "Variable is not reassignable".to_string());
+			report_diagnostic(Diagnostic {
+				message: "Variable is not reassignable".to_string(),
+				span: Some(variable.span()),
+				annotations: vec![DiagnosticAnnotation {
+					message: "Variable is defined here (try adding \"var\" in front)".to_string(),
```

Comment:
I feel like annotations should be read highly contextually and not even require complete sentences. Maybe even no need to start with a capital letter:

```suggestion
					message: "defined here (try adding \"var\" in front)".to_string(),
```

WDYT?